elastic/kibana
# Add execution context support for management plugins

Fixes https://github.com/elastic/kibana/issues/126802

This PR adds execution context support to the platform deployment management apps.

- [x] Ingest Pipelines
- [x] Index Management (Indices, Component Template, Index Template, Data Streams tabs)
- [x] Snapshot & Restore (Snapshots, Repositories, Policies, Restore tabs)
- [x] ILM
- [x] CCR
- [x] Remote clusters
- [x] License Management
- [x] Rollups
- [x] Watcher
- ~Dev Tools (Search Profiler, Console, Painless Lab, Grokdebugger)~ Implemented via https://github.com/elastic/kibana/pull/124996
- ~Upgrade Assistant~ Not applicable, as UA is only enabled in 7.17 and this feature is available in 8.2+

Note: I only added this at the top-level app render, with the exception of Index Management and Snapshot + Restore which has it per tab. If we want, we can be more granular and register each route. I propose the current approach, and modify as needed once we start to analyze the data.

![execution_context1](https://user-images.githubusercontent.com/5226211/160472001-85cbce76-f7b0-4a36-ac21-b337554e92cc.png)

![context_2](https://user-images.githubusercontent.com/5226211/160472016-c5d898dd-9b25-4bce-99b6-7d5f14936c68.png)
![context3](https://user-images.githubusercontent.com/5226211/160472020-9a384b57-3b6a-48ff-8616-d9ccc32d9c7f.png)


### How to test
Follow the "How to test this PR?" instructions in https://github.com/elastic/kibana/pull/124996#issue-1127555423.

Navigate to the list of apps above and verify the events have the correct labels.
